R/test-threads.R

```diff
@@ -330,4 +298,63 @@ delete_models <- function(.mods, .tags = "test threads", .force = FALSE){
 }
 
 
+#' Parse model files and overwrite estimation iterations
+#'
+#' @param .mods a bbi model object or list of model objects. Generally created by `test_threads()`.
+#' @param .max_eval Max number of iterations for NONMEM to run.
+#' @param .detect Estimation options that will be set to `.max_eval` if found.
+#'
+#' @importFrom readr parse_number
+#'
+#' @keywords internal
+adjust_estimation_options <- function(.mods, .max_eval, .detect = c("MAXEVAL", "NITER", "NBURN", "MAX")){
```

Comment:
Should this be more specific than "MAX"? Looking at `estopts` table in nonmem help, seems like there are a good number of names with "MAX" in them.

    ~ % curl -fSsL https://nmhelp.tingjieguo.com/estopts.htm | grep MAX
     ISCALE_MAX                                         X     X        X      X       X
     MAXEVAL                 X
     NUTS_MAXDEPTH                                                                    X
     PSCALE_MAX                                                               X       X

To add to the concern, I don't spot anything in the existing logic that anchors the regular expressions to be more specific or that makes sure only estimation lines are touched.

Reply:
ooh good catch. If you look at `model/nonmem/complex/iovmm.mod` there is an option that's just "MAX":
```
$EST PRINT=5 MAX=9999 Method=1 POSTHOC NOABORT MSF=./10.msf
```

I think we might have to add more complex regex to make sure its separated by a space on both sides or something

Reply:
Although, we might as well make sure all those options are surrounded by a space. No point in making `MAX` unique

Reply:
> I think we might have to add more complex regex 

As you might have guessed from my [comment below](https://github.com/metrumresearchgroup/bbr/pull/542#issuecomment-1213104655), I think you should instead consider just letting the caller give the option they mean.

Reply:
> we might as well make sure all those options are surrounded by a space

You're mostly dealing with them after they are split on spaces, so it's more likely you'll want to anchor (mentioned above).